src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -499,8 +501,9 @@ else if(entry.endsWith("*"))
      * Credentials can be passed e.g. to support running Jenkins behind a (reverse) proxy requiring authorization
      * FIXME: similar to hudson.remoting.Util.openURLConnection which is still used in hudson.remoting.Launcher
      */
+    @Restricted(NoExternalUse.class)
     @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD", justification = "Used by the agent for retrieving connection info from the server.")
-    static URLConnection openURLConnection(URL url, String credentials, String proxyCredentials,
+    public static URLConnection openURLConnection(URL url, String credentials, String proxyCredentials,
```

Comment:
The more widely used copy of the utility code deleted from the `Util` class. Visibility is being increased so that this code can be reused.